bson/src/main/scala/org/mongodb/scala/bson/codecs/Macros.scala

```diff
@@ -83,4 +84,23 @@ object Macros {
   @compileTimeOnly("Creating a Codec utilises Macros and must be run at compile time.")
   def createCodec[T](codecRegistry: CodecRegistry): Codec[T] = macro CaseClassCodec.createCodec[T]
 
+  /**
+   * Annotations to use on case classes that are being processed by macros.
+   */
+  object Annotations {
+
+    /**
+     * Ignores any None values in the case class
+     *
+     * {{{
+     * @IgnoreNone
+     * case class Person(name: String, nickName: Option[String])
+     *
+     * Person(name = "John", nickName: None) // becomes the equivalent of: { name: "John" }
+     * }}}
+     */
+    @meta.param
+    case class IgnoreNone() extends StaticAnnotation
```

Comment:
I'm able to add this annotation to an individual field, e.g.

    case class OptionalValueIgnoreNone(name: String,
                                       @IgnoreNone 
                                       value: Option[String])
 
As that currently has no effect, is there a way to make it a compiler error?